prdoc/pr_7286.prdoc

```diff
@@ -0,0 +1,13 @@
+title: Remove feature flag checks for Elastic Scaling MVP
+doc:
+  - audience: Runtime Dev
+    description: |
+      This PR removes conditional checks for FeatureIndex::ElasticScalingMVP,
+      assuming elastic scaling is always enabled. This simplifies the backing
+      and provisioner logic by making dynamic core allocation the default
```

Comment:
I don't understand what you mean by dynamic core allocation. I'd remove this part

Reply:
Hey @alindima I meant to communicate `Based on demand, automatically occupy or release cores`.  Should i rephrase or just remove it.

Reply:
It's not accurate so I'd remove this bit. The core allocation is handled by agile coretime and acquiring coretime is the parachain's responsibility. It's a different issue

Reply:
Gotcha. Thanks for clarifying on that